File: client/src/app/models/petOwner.ts
export interface PetOwner {
  id: string;
  ownerName: string;
  email: string;
  phone: string;
  pets: Pet[];
}

export interface Pet {
  id?: string;
  petName: string;
  petType: string;
  petAge: string;
}

export class PetOwner implements PetOwner {
  constructor(init?: PetOwnerFormValues) {
    Object.assign(this, init);
  }
}

export class PetOwnerFormValues {
  ownerName: string = "";
  email: string = "";
  phone: string = "";
  pets: Pet[] = [];

  constructor(petOwner?: PetOwner, pets?: Pet[]) {
    if (petOwner && pets) {
      this.ownerName = petOwner.ownerName;
      this.email = petOwner.email;
      this.phone = petOwner.phone;
      this.pets = pets;
      return;
    }

    if (petOwner) {
      this.ownerName = petOwner.ownerName;
      this.email = petOwner.email;
      this.phone = petOwner.phone;
      this.pets = petOwner.pets;
      return;
    }
  }
}

File: Domain/PetOwner.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class PetOwner
    {
        public Guid Id { get; set; }
        public string OwnerName { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public DateTime CreatedOn { get; set; } = DateTime.UtcNow;
        public ICollection<Pet> Pets { get; set; } = new List<Pet>();
        public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    }
}